arm-targets/src/lib.rs

```diff
@@ -0,0 +1,331 @@
+                Arch::Armv7M => "v7-m",
+                Arch::Armv7EM => "v7e-m",
+                Arch::Armv7R => "v7-r",
+                Arch::Armv8R => "v8-r",
+                Arch::Armv8MBase => "v8-m.base",
+                Arch::Armv8MMain => "v8-m.main",
+                Arch::Armv7A => "v7-a",
+                Arch::Armv8A => "v8-a",
+            }
+        )
+    }
+}
+
+/// The Arm Architecture Profile.
+pub enum Profile {
+    /// Microcontrollers
+    M,
+    /// Real-Time
+    R,
+    /// Applications
+    A,
+}
+
+impl Profile {
+    /// Decode a target string
+    pub fn get(target: &str) -> Option<Profile> {
+        let arch = Arch::get(target)?;
+        Some(arch.profile())
+    }
+
+    /// Get a comma-separated list of values, suitable for cfg-check
+    pub fn values() -> String {
+        let string_versions: Vec<String> = [Profile::A, Profile::R, Profile::M]
+            .iter()
+            .map(|i| format!(r#""{i}""#))
+            .collect();
+        string_versions.join(", ")
+    }
+}
+
+impl core::fmt::Display for Profile {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                Profile::M => "m",
+                Profile::R => "r",
+                Profile::A => "a",
+            }
+        )
+    }
+}
+
+/// How does the processor handle multi-byte values
+pub enum Endianness {
+    /// The least-significant byte goes in the lowest numbered address
+    Little,
+    /// The most-significant byte goes in the lowest numbered address
+    Big,
+}
+
+impl Endianness {
+    /// Decode a target string
+    ///
+    /// We check for the known big-endian targets and assume anything else is
+    /// little-endian.
+    pub fn get(target: &str) -> Endianness {
+        if target.starts_with("aarch64be") || target.starts_with("armeb") {
+            Endianness::Big
+        } else {
+            Endianness::Little
+        }
+    }
+
+    /// Get a comma-separated list of values, suitable for cfg-check
+    pub fn values() -> String {
+        let string_versions: Vec<String> = [Endianness::Little, Endianness::Big]
+            .iter()
+            .map(|i| format!(r#""{i}""#))
+            .collect();
+        string_versions.join(", ")
+    }
+}
+
+impl core::fmt::Display for Endianness {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                Endianness::Little => "little",
+                Endianness::Big => "big",
+            }
+        )
+    }
+}
+
+/// How are floating point values passed to functions?
+pub enum FloatAbi {
+    /// Floats are passed using integer registers
+    Soft,
+    /// Floats are passed using FPU registers
+    Hard,
+}
+
+impl FloatAbi {
+    /// Decode a target string
+    pub fn get(target: &str) -> Option<FloatAbi> {
+        if target.ends_with("-eabi") {
```

Comment:
I assume we don't care about, eg, `musleabihf` since these will always be bare metal?

Reply:
yeah, this is really only for bare-metal targets because that's all I need. I'm happy to take PRs to add, e.g. arm64e-apple-darwin, or whatever.